### DISCUSSION TITLE
How to make a default value depend on another in v2?

### DISCUSSION THREAD

**Discussion by ianliu** [pydantic/pydantic#6661] [Question] [answered]
Is it possible to make a field's default value depend on another field value? Something like:

```python
class Event(BaseModel):
    prefix: str = "/tmp"
    ds1: str = "{prefix}/ds1"
    ds2: str = "{prefix}/ds2"

print(Event(prefix="/opt")) # ds1 = "/opt/ds1", ds2 = "/opt/ds2"
print(Event(prefix="/opt", ds2="/var/ds2")) # ds1 = "/opt/ds1", ds2 = "/var/ds2"
```

I though of making this with a `model_validator(mode="after")`, where I walk through all fields, updating them.

Any hints?

**Comment by hennink** [comment, marked as answer]
I think this is covered by the `default_factory` parameter of `Field`, which _'can also take a single required argument, in which the case the already validated data will be passed as a dictionary'_ (see the [docs](https://docs.pydantic.dev/2.10/concepts/fields/#default-values)). They provide the example

```python
from pydantic import BaseModel, EmailStr, Field


class User(BaseModel):
    email: EmailStr
    username: str = Field(default_factory=lambda data: data['email'])


user = User(email='user@example.com')
print(user.username)
#> user@example.com
```
